src/gh/diffCheck/diffCheck/df_error_estimation.py

```diff
@@ -0,0 +1,156 @@
+
+#     return distances
+
+
+def rhino_mesh_2_cloud_distance(source, target, signed=False):
+    """
+        Calculate the distance between every vertex of a Rhino Mesh to its closest point on a PCD
+    """
+    # make a Df point cloud containing all the vertices of the source rhino mesh
+    df_pcd_from_mesh_vertices = diffcheck_bindings.dfb_geometry.DFPointCloud()
+    df_pcd_from_mesh_vertices.points = [[pt.X, pt.Y, pt.Z] for pt in source.Vertices]
+    
+    # calculate the distances
+    distances = np.asarray(df_pcd_from_mesh_vertices.compute_distance(target))
+
+    if signed:
+        # build an RTree containing all the points of the target
+        tree = rg.RTree()
+        for i, ver in enumerate(target.points): 
+            tree.Insert(rg.Point3d(ver[0], ver[1], ver[2]), i)
+
+        for idx, p in enumerate(source.Vertices):
+
+            # find the index on the target that the vertex is closest to
+            search_point = p
+            sphere = rg.Sphere(search_point, distances[idx]*1.0001) #to change later, hack to avoid not finding the point
+            found_indices = []
+
+            def search_callback(sender, e):
+                found_indices.append(e.Id)
+
+            tree.Search(sphere, search_callback)
+
+            df_closest_point = target.points[found_indices[0]]
+            closest_point = rg.Point3d(df_closest_point[0], df_closest_point[1], df_closest_point[2])
+            # Calculate the direction from target to source
+            direction = p - closest_point
+            # Calculate the signed distance
+            normal = source.Normals[idx]
+            dot_product = direction * normal
+            if dot_product < 0:
+                distances[idx] = - distances[idx]
+
+    return np.asarray(distances)
+
+
+def cloud_2_rhino_mesh_distance(source, target, signed=False):
+    """
+        Calculate the distance between every point of a source pcd to its closest point on a target Rhino Mesh
+    """
+
+    #for every point on the point cloud find distance to mesh
+    distances = []
+
+    for p in source.points:
+
+        rhp = rg.Point3d(p[0], p[1], p[2])
+        closest_meshPoint = target.ClosestMeshPoint(rhp, 1000)
+        closest_point = closest_meshPoint.Point
+        distance = rhp.DistanceTo(closest_point)
+
+        if signed:
+            # Calculate the direction from target to source
+            direction = rhp - closest_point
+            # Calculate the signed distance
+            normal = target.NormalAt(closest_meshPoint)
+            dot_product = direction * normal
+            if dot_product < 0:
+                distance = -distance
+
+        distances.append(distance)
+
+    return np.asarray(distances)
+
+
+class DFVizResults:
+    """
+    This class compiles the resluts of the error estimation into one object
+    """
+
+    def __init__(self):
+
+        self.source = []
+        self.target = []
```

Comment:
question: why lists?

Reply:
The idea is that you can input lists of segmented sources and corresponding targets and get back a single Results object. Does this answer the question?